kartothek/serialization/_parquet.py

```diff
@@ -56,9 +76,18 @@ def _reset_dictionary_columns(table):
     2. Massive performance issue due to non-fast path implementation https://issues.apache.org/jira/browse/ARROW-5089
     """
     for i in range(table.num_columns):
-        if pa.types.is_dictionary(table[i].type):
-            new_col = table[i].cast(table[i].data.chunk(0).dictionary.type)
-            table = table.remove_column(i).add_column(i, new_col)
+        col = table[i]
```

Comment:
Wondering if we should skip this entire function for arrow >=0.15.0. At least according to the ticket this has been resolved by now

Reply:
Depends. As far as I understand this, this would mean that the files produced under 0.15.0 would be difficult to be consumed by older versions. What's our policy on that?

Reply:
The performance issue only stems from the arrow->parquet conversion. The issue was that there was no fast-path implemented for dict encoded columns and they needed to be decoded first before they were passed to parquet. 
That's not the critical performance issue but rather that the entire array was decoded for every row group (i.e. 100 row groups -> 100 times decoding instead of only partial decoding or all but only once).


afaik, there is / was never an issue reading them back again.